### PR TITLE
Round `TimeWithTimeZone` when `supportsParametricDateTime` is `false`

### PR DESCRIPTION
**Pull request by mdesmet**
## Description

When `PARAMETRIC_DATETIME` is not set, all temporal types should be rounded up to precision 3. However TimeWithTimeZone is not rounded.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
(x) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Round TimeWithTimeZone values when `PARAMETRIC_DATETIME` is not set
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/server/protocol/QueryResultRows.java [hunk 1]
```diff
@@ -365,6 +367,10 @@ private Object getLegacyValue(Object value, Type type)
                 if (type instanceof TimeType) {
                     return ((SqlTime) value).roundTo(3);
                 }
+
+                if (type instanceof TimeWithTimeZoneType) {
+                    return ((SqlTimeWithTimeZone) value).roundTo(3);
+                }
```
Comment (hashhar) [inline core/trino-main/src/main/java/io/trino/server/protocol/QueryResultRows.java:373]: good catch. I didn't notice this myself when I shared the link. 👍

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTime.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.type;
+
+import org.testng.annotations.Test;
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTime.java:16]: Use junit for new tests.

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java [hunk 3]
```diff
@@ -15,14 +15,66 @@
...
+        assertThat(newInstance(3, 0, 1).toString()).isEqualTo("00:00:00.000+00:01");
+        assertThat(newInstance(4, 0, 1).toString()).isEqualTo("00:00:00.0000+00:01");
+        assertThat(newInstance(5, 0, 1).toString()).isEqualTo("00:00:00.00000+00:01");
+        assertThat(newInstance(6, 0, 1).toString()).isEqualTo("00:00:00.000000+00:01");
+        assertThat(newInstance(7, 0, 1).toString()).isEqualTo("00:00:00.0000000+00:01");
+        assertThat(newInstance(8, 0, 1).toString()).isEqualTo("00:00:00.00000000+00:01");
+        assertThat(newInstance(9, 0, 1).toString()).isEqualTo("00:00:00.000000000+00:01");
+        assertThat(newInstance(10, 0, 1).toString()).isEqualTo("00:00:00.0000000000+00:01");
+        assertThat(newInstance(11, 0, 1).toString()).isEqualTo("00:00:00.00000000000+00:01");
+        assertThat(newInstance(12, 0, 1).toString()).isEqualTo("00:00:00.000000000000+00:01");
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java:38]: Don't convert to String before comparing. Compare directly against instances created with `SqlTimeWithTimeZone.newInstance`

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java [hunk 4]
```diff
@@ -15,14 +15,66 @@
 
 import org.testng.annotations.Test;
 
+import static io.trino.spi.type.SqlTimeWithTimeZone.newInstance;
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java:18]: Don't static import this method. The name is too ambiguous.

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTime.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+import static io.trino.spi.type.SqlTime.newInstance;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestSqlTime
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTime.java:21]: This test is unrelated to the fix in this commit. Move it to a separate commit.

**Review by martint — CHANGES_REQUESTED** [review]

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java (hunk 3 above)
Comment (mdesmet) [reply core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java:38]: I agree in case of all other tests except here we don't apply any transformation, so we have to compare against another representation.

Comment (martint) [reply core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java:38]: Ah, yes, I meant to put the comment in the methods below. For this one, rename the test to `testToString`, as that's what's being tested here.

Also, this test is unrelated to the fix in this commit, so move it to a separate commit.

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTime.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+{
+    @Test
+    public void testBaseline()
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTime.java:24]: Rename to `testToString`

File: core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java [hunk 7]
```diff
@@ -40,7 +40,40 @@ public void testToString()
     @Test
     public void testEqualsDifferentZone()
     {
-        assertThat(SqlTimeWithTimeZone.newInstance(12, 0, 0))
-                .isNotEqualTo(SqlTimeWithTimeZone.newInstance(12, 0, 1));
+        assertThat(SqlTimeWithTimeZone.newInstance(12, 0, 0)).isNotEqualTo(SqlTimeWithTimeZone.newInstance(12, 0, 1));
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java:43]: Unrelated change

Comment (mdesmet) [reply core/trino-spi/src/test/java/io/trino/spi/type/TestSqlTimeWithTimeZone.java]: Sorry for that, removed!
